packages/vite/src/node/server/moduleGraph.ts

```diff
@@ -145,8 +152,13 @@ export class ModuleGraph {
     }
     // Don't invalidate mod.info and mod.meta, as they are part of the processing pipeline
     // Invalidating the transform result is enough to ensure this module is re-processed next time it is requested
-    mod.transformResult = null
-    mod.ssrTransformResult = null
+    if (softInvalidate && mod.softInvalidated !== false) {
```

Comment:
The second part if strange, soft deleting a module twice is hard delete

Reply:
Oh yeah, I was confused, but in that case I would:
- init softInvalidated and removed the null type
- remove this no-op branch